test/unit-test/FreeRTOS_ND/ut.cmake

```diff
@@ -0,0 +1,116 @@
+# Include filepaths for source and include.
+include( ${MODULE_ROOT_DIR}/test/unit-test/TCPFilePaths.cmake )
+
+# ====================  Define your project name (edit) ========================
+set( project_name "FreeRTOS_ND" )
+message( STATUS "${project_name}" )
+
+# =====================  Create your mock here  (edit)  ========================
+set(mock_list "")
+
+# list the files to mock here
+list(APPEND mock_list
+            "${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel/include/task.h"
+            "${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel/include/list.h"
+            "${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel/include/queue.h"
+            "${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel/include/event_groups.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_ARP.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_IP.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_IP_Private.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_IPv6.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_ICMP.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_IP_Timers.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_IP_Utils.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_IPv6_Utils.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_Routing.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_Sockets.h"
+            "${CMAKE_BINARY_DIR}/Annexed_TCP/NetworkBufferManagement.h"
+        )
+
+set(mock_include_list "")
+# list the directories your mocks need
+list(APPEND mock_include_list
+            .
+            ${TCP_INCLUDE_DIRS}
+            ${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel/include
+            ${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel/portable/ThirdParty/GCC/Posix
+            ${MODULE_ROOT_DIR}/test/unit-test/ConfigFiles
+            ${MODULE_ROOT_DIR}/test/unit-test/${project_name}
+        )
+
+set(mock_define_list "")
+#list the definitions of your mocks to control what to be included
+list(APPEND mock_define_list
+            ""
+       )
+
+# ================= Create the library under test here (edit) ==================
+
+set(real_source_files "")
+
+# list the files you would like to test here
+list(APPEND real_source_files
+            ${CMAKE_BINARY_DIR}/Annexed_TCP_Sources//${project_name}.c
+	)
```

Comment:
${CMAKE_BINARY_DIR}/Annexed_TCP_Sources/${project_name}.c

Reply:
An extra '/ ' in the path of the project file
can be changed to ${CMAKE_BINARY_DIR}/Annexed_TCP_Sources/${project_name}.c